wagtail/admin/views/scheduled_pages.py

```diff
@@ -0,0 +1,110 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.shortcuts import get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.http import url_has_allowed_host_and_scheme
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin import messages
+from wagtail.admin.ui.components import Component
+from wagtail.core.models import Page, UserPagePermissionsProxy
+
+
+def get_scheduled_pages_for_user(request):
+    user_perms = UserPagePermissionsProxy(request.user)
+    pages = (
+        Page.objects.annotate_approved_schedule()
+        .filter(_approved_schedule=True)
+        .prefetch_related("content_type")
+        .order_by("-first_published_at")
+        & user_perms.publishable_pages()
+    )
+
+    if getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+        pages = pages.select_related("locale")
+    return pages
+
+
+class ScheduledPagesPanel(Component):
+    name = "scheduled_pages"
+    template_name = "wagtailadmin/scheduled_pages/panel.html"
+    order = 200
+
+    def get_context_data(self, parent_context):
+        request = parent_context["request"]
+        context = super().get_context_data(parent_context)
+        context["pages_to_be_scheduled"] = get_scheduled_pages_for_user(request)
```

Comment:
given this appears on the homepage I would very much like to limit this to something manageable. on large, active sites this could be quite the list.

We do limit the recent edits panel, the others not so much, so need to open an issue (or even PR to sort that out)

Reply:
I'll limit it to 5 on the homepage for now, perhaps a setting should be created for this like in the recent edits panel.